pkg/controllers/follower/controller.go

```diff
@@ -94,38 +91,33 @@ var (
 
 // Handles for a leader or follower type.
 type typeHandles struct {
-	// <leader|follower>+federatedKind
-	name        string
-	typeConfig  *fedcorev1a1.FederatedTypeConfig
-	sourceGK    schema.GroupKind
-	federatedGK schema.GroupKind
-	informer    informers.GenericInformer
-	client      dynamic.NamespaceableResourceInterface
-	worker      worker.ReconcileWorker
+	// <leader|follower>+sourceKind
+	name       string
+	typeConfig *fedcorev1a1.FederatedTypeConfig
+	sourceGK   schema.GroupKind
 }
 
 type Controller struct {
-	name string
-
-	eventRecorder record.EventRecorder
-	metrics       stats.Metrics
-	logger        klog.Logger
-
 	// The following maps are written during initialization and only read afterward,
 	// therefore no locks are required.
 
-	// map from source GroupKind to federated GroupKind
-	sourceToFederatedGKMap map[schema.GroupKind]schema.GroupKind
-	// map from leader federated GroupKind to typeHandle
+	// map from leader object GroupKind to typeHandle
 	leaderTypeHandles map[schema.GroupKind]*typeHandles
```

Comment:
If we keep these handlesMap, they would need to be protected by locks because concurrent read-write is not possible.